test/spatial/diffusion.jl

```diff
@@ -10,51 +11,74 @@ function get_mean_sol(jump_prob, Nsims, saveat)
     sol/Nsims
 end
 
+# assume sites are labeled from 1 to num_sites(spatial_system)
+function discrete_laplacian_from_spatial_system(spatial_system, hopping_rate)
+    sites = 1:DiffEqJump.num_sites(spatial_system)
+    laplacian = zeros(Int, length(sites), length(sites))
+    for site in sites
+        laplacian[site,site] = -DiffEqJump.num_neighbors(spatial_system, site)
+        for nb in DiffEqJump.neighbors(spatial_system, site)
+            laplacian[site, nb] = 1
+        end
+    end
+    laplacian*hopping_rate
+end
+
+# problem setup
 reactstoch = []
 netstoch = []
 rates = []
 majumps = MassActionJump(rates, reactstoch, netstoch)
 tf = 0.5
 u0 = [100]
+num_species = 1
 
 domain_size = 1.0 #μ-meter
-linear_size = 33
+linear_size = 5
 diffusivity = 0.1
-dim = 1
-grid = DiffEqJump.CartesianGrid(dim, linear_size)
-num_nodes = DiffEqJump.num_sites(grid)
+dim = 2
+dims = Tuple([linear_size for i in 1:dim])
+num_nodes = prod(dims)
 
 # Starting state setup
 starting_state = zeros(Int, length(u0), num_nodes)
```

Comment:
At the moment this would break things. It would be easy to support receiving input in this form. In order to support output in this form, there can be a `cartesianize`, which takes a solution with "unrolled" shape and reshapes it into Cartesian shape.